pkg/test/utils.go

```diff
@@ -165,7 +169,42 @@ func CompareGoldenObject(t *testing.T, path string, got []byte) {
 	if err != nil {
 		t.Errorf("Failed parsing file: %s", err)
 	}
-	if diff := cmp.Diff(wantMap, gotMap); diff != "" {
+	if diff := cmp.Diff(postWriteNormalization(wantMap), gotMap); diff != "" {
 		t.Errorf("unexpected diff in %s: %s", path, diff)
 	}
+	if err := os.WriteFile(path+"_2d.yaml", got, 0644); err != nil {
+		t.Errorf("failed to write golden output %s: %v", path, err)
+	}
+}
+
+func postWriteNormalization(wantMap map[string]interface{}) map[string]interface{} {
```

Comment:
Not a fan of this but short of re-capturing to exclude the annotations (which I don't think we should do just yet) I thought this was the lesser evil 🤷🏼

Reply:
I think either we accept that the golden outputs are specific to a particular scenario (which highlights the gap) or we have two files (i.e. add a "scenario" into the filename / path).  I think I'd start with accepting the delta, and then if we can't close the gap, we can add it into the filename / path.